### ISSUE TITLE
Perfume-Genie-ESPhome

### ISSUE THREAD

**Issue by rici44** [Echnics/Perfume-Genie-ESPhome#1]
Hi there. What kind of NFC hardware does the board have? Did you manage NFC to work?

**Comment by Echnics** [comment]
Hey :)
No, i have not looked at that anymore.
And even if i did find out the communication and the pins to that chip, i think  it will be difficult to do much with the data from the tags.
And all you can get is the name of the fragrance,,, and the time you put it inn.
The tank level thing is just based on time estimate when they think it's empty.

Not saying i not would try more if i got any tips on how it was wired or if i knew if it uses i2c,spi or uart connection.

**Comment by InsaneSoftware** [comment]
As one of the creators of the genie i like theze kind of hobby projects =)

We use the mfrc522 libary in our source code to read scents and scent usage on the tag.

On page 4 of the nfid tag is a hexidecimal of the scent. I can give you the list to match hexidecimals to scents ^^

and yes we use an uart cable to read data on thoze pinns but its not that easy.. we secured it...

I cant tell you how to bypass our security of the uart pins due contract limitations but maybe you will find out yourself ^^

**Comment by Echnics** [comment]
That would be great :D
I'll take all the info you have and can give.
Okey, so there some security thing on the UART?  But can you say what pins on the esp it uses?

**Comment by rici44** [comment]
@InsaneSoftware thanks for the info you can share. It would be great if you may share also that formula for cartridge usage. I guess it's based of measured evaporation of the perfume, isn't it?

**Comment by InsaneSoftware** [comment]
I cannot share formulas but can help you about fill status BUT:

**_Cartridge usage is not safe to use its only a bold indication_** (thats wy we removed it from the app)

Still if you want to use this value.. this is also stored on the NFC tag

Here a little table for you:

![IFIDs83](https://github.com/user-attachments/assets/ebd467b7-9c1f-4a4a-b435-3ea630dcc028)

**Comment by TomStarren** [comment]
@InsaneSoftware can you share which pins you are using for the RFID reader and is UART the only way to go? As far as I know rc522 doesn't work anymore on UART in ESPHome

Also the hexadecimals for the scent is also helpfull.

**Comment by domasist** [comment]
Will this work also for Genie 1?

**Comment by InsaneSoftware** [comment]
> Will this work also for Genie 1?

Yes it only works for older devices ^^ new devices are encrypted due the stupid Cyber security law.

Dont forget to keep it local and not connect to our servers with custom firmware ^^ (you will get blocked)

sorry @TomStarren  i almost forget this repo but now others are talking from home assistant about this.. better late then sorry heres the list for your LOCAL scents:

> 
> The Ritual of Dao	044616f0
> The Ritual of The Happy Buddha	04c61750
> The Ritual of Sakura	053616b0
> The Ritual of Ayurveda	4179750
> The Ritual of Spring Garden	5370720
> The Ritual of Hammam	048616d0
> The Ritual of Karma	06b61720
> Private Collection Goji Berry	0476f6a0
> Private Collection Oriental Vetiver	04f72690
> Private Collection Black Oudh	0426c610
> Private Collection Precious Amber	5072650
> Private Collection Sweet Jasmine	5377650
> Private Collection Imperial Rose	0496d700
> Private Collection Savage Garden	5361760
> Cartridge tag invalid	013a0c
> The Ritual of Jing	04a696e0
> The Ritual of Mehr	06d65680
> Private Collection Suede Vanilla	056616e0
> Private Collection Cotton Blossom	0426c6f0
> Private Collection Green Cardamom	4772650
> Private Collection Royal Tea	4772750
> The Ritual of Jing Night	4773750
> HIDDEN NEW SCENT	4774750
> HIDDEN NEW SCENT	02d63620

I cannot tell you the new scents ^^ wait for official release ^^

Im not allowed to tell you about the pinning etc (Sorry)
